### PR TITLE
Add public_key_algorithm_oid to certificate and csr

### PR DESCRIPTION
**Pull request by Julien00859**

### CODE REVIEW

**Comment by Julien00859** [comment]
The two new certificate and public key for the unittest were created as follow:

    openssl genpkey -out /tmp/rsa_key.pem -algorithm RSA
    openssl req -x509 -days 7000 \
                -key /tmp/rsa_key.pem -out rsae_cert.pem \
                -sigopt rsa_padding_mode:pss -sha256 -sigopt rsa_pss_saltlen:20 \
                -subj "/CN=cryptography.io"
    openssl rsa -in /tmp/ca-rsa.key.pem -pubout -outform DER > ../../asymmetric/PKCS8/rsae_2048_pub.der

File: vectors/cryptography_vectors/x509/custom/rsae_cert.pem [hunk 1]
```diff
@@ -0,0 +1,21 @@
+-----BEGIN CERTIFICATE-----
```
Comment (alex) [inline vectors/cryptography_vectors/x509/custom/rsae_cert.pem:1]: Any new vectors need to be documented in `test-vectors.md`.

File: src/cryptography/hazmat/_oid.py [hunk 2]
```diff
@@ -245,6 +256,14 @@ class AttributeOID:
...
         "GOST R 34.10-2012 with GOST R 34.11-2012 (512 bit)"
     ),
+    PublicKeyAlgorithmOID.DSA: "dsaEncryption",
+    PublicKeyAlgorithmOID.EC_PUBLIC_KEY: "id-ecPublicKey",
+    PublicKeyAlgorithmOID.RSAES_PKCS1_v1_5: "rsaEncryption",
+    PublicKeyAlgorithmOID.RSASSA_PSS: "rsassaPss",
+    PublicKeyAlgorithmOID.X25519: "X25519",
+    PublicKeyAlgorithmOID.X448: "X448",
+    PublicKeyAlgorithmOID.ED25519: "ED25519",
+    PublicKeyAlgorithmOID.ED448: "ED448",
```
[the remark this answers is not in the document]
Comment (Julien00859) [reply src/cryptography/hazmat/_oid.py]: I made sure there was no other duplicate, only ed25519 and ed448 were duplicated. Is this mapping also used value->key?

File: vectors/cryptography_vectors/x509/custom/rsae_cert.pem (hunk 1 above)
Comment (Julien00859) [reply vectors/cryptography_vectors/x509/custom/rsae_cert.pem]: Oh nice, thanks to this documentation I could reuse another RSA key for my test. Only needed to add a single file, which is now documented.

**Comment by Julien00859** [comment]
> [force-pushed](https://github.com/pyca/cryptography/compare/e3484edeec9283e068e755979f09a7069bfbe160..91e081710439aa09c2668b506702895b313d05f3) the public_key_oid branch from[ e3484ed ](https://github.com/pyca/cryptography/commit/e3484edeec9283e068e755979f09a7069bfbe160)to 91e0817

changes following the review

> [Julien00859](https://github.com/Julien00859) [force-pushed](https://github.com/pyca/cryptography/compare/91e081710439aa09c2668b506702895b313d05f3..ac42a707e702e09dcf588ff5d0f7ccf96ab2ef1a) the public_key_oid branch from[ 91e0817 ](https://github.com/pyca/cryptography/commit/91e081710439aa09c2668b506702895b313d05f3)to ac42a70

rebase to solve the conflict inside `CHANGELOG.rst`

**Comment by reaperhulk** [comment]
Looks like you need to add PSS to spelling_wordlist.txt

**Review by reaperhulk — APPROVED** [review]

**Comment by reaperhulk** [comment]
Thanks for the contribution!

**Pull request merged**